packages/devtools_app/lib/src/service/resolved_uri_manager.dart

```diff
@@ -0,0 +1,51 @@
+import '../shared/globals.dart';
+import 'vm_service_wrapper.dart';
+
+/// Manager for handling package Uri lookup and caching.
+class ResolvedUriManager {
+  Map<String, String?>? _resolvedUrlMap;
+  VmServiceWrapper? service;
+
+  /// Initializes the [ResolvedUriManager]
+  void vmServiceOpened() {
+    _resolvedUrlMap = <String, String?>{};
+  }
+
+  /// Cleans up the resources of the [ResolvedUriManager]
+  void vmServiceClosed() {
+    _resolvedUrlMap = null;
+  }
+
+  /// Calls out to the [VmService] to lookup unknown uri to package uri mappings.
+  ///
+  /// Known uri mappings are cached to avoid asking [VmService] for the same
+  /// mapping.
+  ///
+  /// [uris] List of uris to fetch package uris for.
+  Future<void> fetchPackageUris(
+    String isolateId,
+    List<String> uris,
```

Comment:
nit: I think this can fit on one line. If I'm wrong, add back the trailing comma